event_routing_backends/processors/xapi/event_transformers/completion_events.py

```diff
@@ -63,6 +65,23 @@ class LessonCompletionTransformer(ModuleCompletionTransformer):
     """
     object_type = constants.XAPI_ACTIVITY_LESSON
 
+    def transform(self):
+        """
+        Return transformed `Statement` object.
+
+        `BaseTransformer`'s `transform` method will return dict containing
+        xAPI objects in transformed fields. Here we return a `Statement` object
+        constructed using those fields.
+
+        Returns:
+            `Statement`
+        """
+        block_id = self.get_data("data.block_id", required=True)
+        usage_key = UsageKey.from_string(block_id)
+        vertical = modulestore().get_item(usage_key)
+
+        return {} if vertical.graded else super().transform()
```

Comment:
Here also continues but later raises the exception in the worker of `eventtracking.processors.exceptions.EventEmissionExit was raised`?. I'm right?
? There is no way to break the flow without exception. Like a break or function or feature flag to avoid the final send?

Reply:
why do you say that, if you don't the exception that I said ? hahah
No, that is the event-routing -backends way

Reply:
Maybe could be a step with a field in the transform dict `like avoid_send`, but that is more code, more features, and is not part of this scope.
Let's go with this approach and analyze the quantity of exceptions raised of this kind in the future.